testcases/MachineConfig.py

```diff
@@ -0,0 +1,269 @@
+            except AttributeError:
+                self.pmem_name = "vol1"
+            try: self.pmem_size = conf.args.pmem_size
+            except AttributeError:
+                self.pmem_size = "8192"
+            curr_num_volumes = self.cv_HMC.vpmem_count()
+            if curr_num_volumes[0] >= "1":
+                log.info("System already have vpmem volume.")
+            else:
+                current_lmb = self.cv_HMC.get_lmb_size()
+                if int(self.pmem_size) % int(current_lmb[0]) != 0:
+                    self.fail("pmem_size should be multiple of %s" % current_lmb)
+                self.cv_HMC.configure_vpmem(self.pmem_name, self.pmem_size)
+                curr_num_volumes = self.cv_HMC.vpmem_count()
+                if curr_num_volumes[0] >= "1":
+                    log.info("Configured vpmem %s of %sMB" % (self.pmem_name, self.pmem_size))
+                else:
+                    self.fail("Failed to configure pmem")
+
+
+        self.cv_HMC.run_command("chsysstate -r lpar -m %s -o on -n %s -f %s" %
+                               (self.system_name, self.lpar_name, self.lpar_prof))
+        time.sleep(5)
+        curr_proc_mode = self.cv_HMC.get_proc_mode()
+        if proc_mode in curr_proc_mode:
+            log.info("System booted with %s mode" % proc_mode)
+        else:
+            self.fail("Failed to boot in %s mode" % proc_mode)
+
+
+class RestoreLAPRConfig(MachineConfig):
+
+    '''
+    This class boots the lpar with previous profile, this can be called after ChangeToSharedMode or ChangeToDedicatedMode.
+    '''
+
+    def setUp(self):
+        super(RestoreLAPRConfig, self).setUp()
+        conf = OpTestConfiguration.conf
+
+    def runTest(self):
+        self.cv_HMC.poweroff_lpar()
+        self.cv_HMC.lpar_prof = self.cv_HMC.lpar_prof + "_bck"
+        self.cv_HMC.poweron_lpar()
+
+
+class CecConfig(MachineConfig):
+
+    '''
+    This class configures LMB. Pass lmb_size to cec_config in config file.
+    Ex: cec_config="lmb=4GB, hugepage=16GB"
+    Valid LMB values are "128,256,1024,2048,4096"
+    '''
+
+    def setUp(self):
+        super(CecConfig, self).setUp()
+        conf = OpTestConfiguration.conf
+        try: self.cec_config = conf.args.cec_config
+        except AttributeError:
+            self.cec_config="lmb=256"
+
+    def runTest(self):
+        if not self.cv_HMC.lpar_vios:
+            self.skipTest("Please pass lpar_vios in config file.")
+        if "lmb=4GB" in self.cec_config or "lmb=4096" in self.cec_config:
+            self.lmb_size=4096
+        elif "lmb=2GB" in self.cec_config or "lmb=2048" in self.cec_config:
+            self.lmb_size=2048
+        elif "lmb=1GB" in self.cec_config or "lmb=1024" in self.cec_config:
+            self.lmb_size=1024
+        elif "lmb=256MB" in self.cec_config or "lmb=256" in self.cec_config:
+            self.lmb_size=256
+        elif "lmb=128MB" in self.cec_config or "lmb=128" in self.cec_config:
+            self.lmb_size=128
```

Comment:
@Pavithra1602 let the user pass the input in only MBs, that will avoid above code..  make sure the user provides input in 1024, 2048, 128, 256   as input... and document ins lmb readme/doc string asking user to provide all sizes in MBs 

also maintain a list of supported sizes  valid_size = [128, 256, 1024, 2048]  and  just a `if size in valid size:`  check will  do

Reply:
yes good idea @abdhaleegit , @Pavithra1602 can you please adopt above suggestion